sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/Transforms.java

```diff
@@ -67,9 +70,9 @@ static TextDocumentBatchStatistics toBatchStatistics(RequestStatistics statistic
     static com.azure.ai.textanalytics.models.TextAnalyticsError toTextAnalyticsError(
         TextAnalyticsError textAnalyticsError) {
         return new com.azure.ai.textanalytics.models.TextAnalyticsError(
-            ErrorCodeValue.fromString(textAnalyticsError.getCode().toString()), textAnalyticsError.getMessage(),
-            textAnalyticsError.getTarget(), textAnalyticsError.getDetails() == null ? null
-            : setErrors(textAnalyticsError.getDetails()));
+            TextAnalyticsErrorCode.fromString(textAnalyticsError.getInnerError().getCode().toString()),
```

Comment:
I might be missing something, but what happens if no innerError is returned? Will the top level error code/message/target be used?

Reply:
It currently doesn't do that, but I think it should. Will update!